main/src/ca/uwaterloo/flix/language/ast/WeededAst.scala

```diff
@@ -299,12 +299,16 @@ object WeededAst {
 
     case class RecordExtend(label: Name.Ident, tpe: WeededAst.Type, rest: WeededAst.Type, loc: SourceLocation) extends WeededAst.Type
 
+    case class RecordGeneric(tvar: WeededAst.Type.Var, loc: SourceLocation) extends WeededAst.Type
```

Comment:
I assume this exists to "remember" that we know something is a record variable?

What about "Boolean type variables", e.g. for effects?

Reply:
Yes, that's right.

The parser previously parsed `{| r}` and `r` the same (as `WeededAst.Type.Var(...)`), so the context was lost after parsing.

Boolean types don't need this since we don't have the same loss in parsing. `WeededAst.Type.Arrow.eff` has to be of Boolean kind.